libs/wasm/transform/src/lib.rs

```diff
@@ -130,74 +143,111 @@ fn generate_transform_request(wasm_request: &WASMRequest) -> Result<transform::R
             }
         }
         TransformType::TRANSFORM_TYPE_DELETE_FIELD => {
-            // Never had deprecated options, no need to handle here
+            let opts = t.delete_field_options();
 
-            if path.is_empty() {
-                path = t.delete_field_options().path.clone();
+            // No inter-step result, pull path from request
+            if paths.is_empty() {
+                paths.push(DetectiveStepResultMatch {
+                    path: opts.path.clone(),
+                    value: "".to_string().into_bytes(),
+                    type_: EnumOrUnknown::new(DETECTIVE_TYPE_UNKNOWN),
+                    special_fields: Default::default(),
+                })
             }
 
             transform::Request {
                 data: wasm_request.input_payload.clone(),
-                path: path,
                 value: "".to_string(),
+                paths: paths,
                 truncate_options: None,
                 extract_options: None,
             }
         }
         TransformType::TRANSFORM_TYPE_MASK_VALUE => {
-            if path.is_empty() {
-                path = t.mask_options().path.clone();
+            let opts = t.mask_options();
+
+            // No inter-step result, pull path from request
+            if paths.is_empty() {
+                paths.push(DetectiveStepResultMatch {
+                    path: opts.path.clone(),
+                    value: "".to_string().into_bytes(),
+                    type_: EnumOrUnknown::new(DETECTIVE_TYPE_UNKNOWN),
+                    special_fields: Default::default(),
+                })
             }
 
             transform::Request {
                 data: wasm_request.input_payload.clone(),
-                path: path,
-                value: "".to_string(),
+                paths: paths,
+                value: opts.mask.clone(),
                 truncate_options: None,
                 extract_options: None,
             }
         }
         TransformType::TRANSFORM_TYPE_OBFUSCATE_VALUE => {
-            if path.is_empty() {
-                path = t.obfuscate_options().path.clone();
+            let opts = t.obfuscate_options();
+
+            // No inter-step result, pull path from request
+            if paths.is_empty() {
+                paths.push(DetectiveStepResultMatch {
+                    path: opts.path.clone(),
+                    value: "".to_string().into_bytes(),
```

Comment:
Really wish there was a way to avoid having to specify these defaults every time. Is there some way to instantiate a struct AND have it set defaults? 🙏

Reply:
Fixed with `..Default::default()` :)